pyccel/parser/semantic.py

```diff
@@ -3486,25 +3442,26 @@ def _insert_obj(location, target, obj):
                 if new_name != old_name:
                     import_free = import_free.clone(new_name)
 
-            expr = Import(source, targets)
+            mod = p.semantic_parser.ast
 
             if __import_all__:
-                mod = p.semantic_parser.ast
-                expr = Import(source_target, AsName(mod, __module_name__))
+                expr = Import(source_target, AsName(mod, __module_name__), mod=mod)
                 container['imports'][source_target] = expr
 
             elif __module_name__:
-                expr = Import(__module_name__, targets)
+                expr = Import(__module_name__, targets, mod=mod)
                 container['imports'][source_target] = expr
 
             elif not __ignore_at_import__:
+                expr = Import(source, targets, mod=mod)
                 container['imports'][source_target] = expr
 
         return result
 
 
 
     def _visit_With(self, expr, **settings):
+        scope = self.create_new_loop_scope()
```

Comment:
A With block is not a loop, but it requires a scope which allows variables to escape it in python, like loops. These scopes need a better name

Reply:
Do we have any example of pyccelizable code that contains the `With` statement?

Reply:
The code is not fully pyccelizable but there is one test [here](https://github.com/pyccel/pyccel/blob/master/tests/codegen/fcode/scripts/context.py#L40)